namer/namer.cc

```diff
@@ -15,16 +18,1336 @@
 
 using namespace std;
 
+// TODO:
+// - Remove redundancy whenever possible.
+// - ENFORCE that NameFinder never produces errors.
+
 namespace sorbet::namer {
 
+namespace {
+class FoundNames;
+
+struct FoundClassRef;
+struct FoundClass;
+struct FoundStaticField;
+struct FoundTypeMember;
+struct FoundMethod;
+
+enum class NameKind : u1 {
+    None = 0,
+    Root = 1,
+    Class = 2,
+    ClassRef = 3,
+    Method = 4,
+    StaticField = 5,
+    TypeMember = 6,
+    Symbol = 7,
+};
+
+class FoundNameRef final {
```

Comment:
I don't quite get why this is called `FoundNameRef`. It doesn't look like it stores `NameRef`s.

Is there another word we can invent for the kinds of data structures being tracked in this file? Like maybe `Definition`s?